packages/2d/src/components/SVG.ts

```diff
@@ -0,0 +1,387 @@
+      deleted: [],
+      transformed: [],
+    };
+
+    const aNodes = from.nodes;
+    const bNodes = to.nodes;
+    const aLength = aNodes.length;
+    const bLength = bNodes.length;
+    let aIndex = 0;
+    let bIndex = 0;
+
+    diffSequence(
+      aLength,
+      bLength,
+      (aIndex, bIndex) => {
+        return this.isNodeEqual(aNodes[aIndex], bNodes[bIndex]);
+      },
+      (nCommon, aCommon, bCommon) => {
+        if (aIndex !== aCommon)
+          diff.deleted.push(...aNodes.slice(aIndex, aCommon));
+        if (bIndex !== bCommon)
+          diff.inserted.push(...bNodes.slice(bIndex, bCommon));
+
+        aIndex = aCommon;
+        bIndex = bCommon;
+        for (let x = 0; x < nCommon; x++) {
+          diff.transformed.push({
+            from: aNodes[aIndex],
+            to: bNodes[bIndex],
+          });
+          aIndex++;
+          bIndex++;
+        }
+      },
+    );
+
+    if (aIndex !== aLength) diff.deleted.push(...aNodes.slice(aIndex));
+
+    if (bIndex !== bNodes.length) diff.inserted.push(...bNodes.slice(bIndex));
+
+    diff.deleted = diff.deleted.filter(aNode => {
+      const bIndex = diff.inserted.findIndex(bNode =>
+        this.isNodeEqual(aNode, bNode),
+      );
+      if (bIndex >= 0) {
+        const bNode = diff.inserted[bIndex];
+        diff.inserted.splice(bIndex, 1);
+        diff.transformed.push({
+          from: aNode,
+          to: bNode,
+        });
+
+        return false;
+      }
+
+      return true;
+    });
+    return diff;
+  }
+
+  private cloneNodeExact(node: Node) {
+    const props: ShapeProps = {
+      position: node.position(),
+      scale: node.scale(),
+      rotation: node.rotation(),
+    };
+    if (node instanceof Layout) {
+      props.size = node.size();
+    }
+    return node.clone(props);
+  }
+
+  public *tweenSVG(svg: string, time: number) {
+    const newSVG = this.parseSVG(svg);
+    const diff = this.diffSVG(this.parsed(), newSVG);
+    const transformed = diff.transformed.map(({from, to}) => ({
+      from: this.cloneNodeExact(from),
+      current: from,
+      to,
+    }));
+
+    for (const node of diff.inserted) this.wrapper.add(node);
+
+    const autoWidth = this.customWidth() == null;
+    const autoHeight = this.customHeight() == null;
+
+    const beginning = 0.2;
+    const ending = 0.8;
+    const overlap = 0.15;
+
+    yield* tween(
+      time,
+      value => {
+        const remapped = clampRemap(beginning, ending, 0, 1, value);
+        const eased = easeInOutSine(remapped);
+        for (const node of transformed) {
+          node.current.position(
+            Vector2.lerp(node.from.position(), node.to.position(), eased),
+          );
+          node.current.scale(
+            Vector2.lerp(node.from.scale(), node.to.scale(), eased),
+          );
+          node.current.rotation(
+            easeInOutSine(remapped, node.from.rotation(), node.to.rotation()),
+          );
+
+          if (node.current instanceof Rect) {
```

Comment:
Shouldn't it be `Layout`?
```suggestion
          if (node.current instanceof Layout) {
```

Reply:
It must be `Shape` not `Rect` or `Layout` because all node is actually `Shape`.

Reply:
The if statement actually usable. I have been rewrite this.